### PR TITLE
[MNT] Updated `compatibility.yaml` file and e2e tests

### PR DESCRIPTION
**Pull request by rmanaem**
Changes proposed in this pull request:

- Updated e2e tests
- Updated `compatibility.yaml` file
  -  Added a step to modify the `NB_API_QUERY_URL` env var before bringing the stack up
  - Renamed the workflow from `Compatibility of` to `Tool version compatibility test`
- Added badge for `Tool version compatibility test` to README.md

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [ ] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Review by surchs — APPROVED** [review]
Looks great :cook:

File: README.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
+<div align="center">
+
 # recipes
 Configuration files for a Neurobagel deployment.
 
+<a href="https://github.com/neurobagel/recipes/actions/workflows/compatibility.yaml">
+        <img src="https://img.shields.io/github/actions/workflow/status/neurobagel/query-tool/component-test.yaml?color=8FBC8F&label=Tool version compatibility test&style=flat-square" alt="Tool version compatibility test">
+    </a>
+</div>
+
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply README.md:10]: Unfortunately no, the status badge references a specific workflow file and can't be made for particular jobs in that file.
The workaround would be to make another workflow for `nightly`, what do you think?

[the remark this answers is not in the document]
Comment (surchs) [reply README.md:10]: Hhm. For now let's stick with one. But then let's rename the workflow to "Version compatibility" or something that makes sense on the badge alone. I.e. "Compatibility of: failing" is not very informative

**Comment by surchs** [comment]
We should also add an internal release label

**Comment by rmanaem** [comment]
> We should also add an internal release label

Do u mean to release off of this PR?

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmanaem) [reply README.md:10]: It's already been renamed, the badge should show `Tool version compatibility test`

**Pull request merged**